core/storage/pager.rs

```diff
@@ -208,6 +207,10 @@ pub struct Pager {
     pub buffer_pool: Arc<BufferPool>,
     /// I/O interface for input/output operations.
     pub io: Arc<dyn crate::io::IO>,
+    /// Indicates whether the database is running as transient or persistent
+    #[cfg_attr(feature = "omit_autovacuum", allow(dead_code))]
+    database_mode: DatabaseMode,
+
```

Comment:
Couldn't this be a function from I/O trait?

Reply:
added the function to the I/O trait in 5ffc70819941082df3ce683cd28f04c7f14706ff